addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -195,9 +195,7 @@ def _compute_description_validity(self):
     def _compute_leaves(self):
         for allocation in self:
             allocation.max_leaves = allocation.number_of_hours_display if allocation.type_request_unit == 'hour' else allocation.number_of_days
-            allocation.leaves_taken = sum(taken_leave.number_of_hours_display if taken_leave.leave_type_request_unit == 'hour' else taken_leave.number_of_days\
-                for taken_leave in allocation.taken_leave_ids\
-                if taken_leave.state == 'validate')
+            allocation.leaves_taken = allocation.holiday_status_id._get_employees_days_per_allocation([allocation.employee_id.id])[allocation.employee_id.id][allocation.holiday_status_id][allocation]['leaves_taken']
```

Comment:
By looking at the way ``_get_employees_days_per_allocation`` is defined, I think you could batch the calls per ``holiday_status_id`` right ?  :)

Reply:
And then iterate on the allocation as you do right now to retrieve the result for the related employee